coast/DOMAIN.py

```diff
@@ -110,3 +112,33 @@ def subset_indices_by_distance(self, centre_lon, centre_lat, radius):
         # Then these output tuples can be separated into x and y arrays if necessary.
 
         return nemo_indices
+
+    def find_J_I(self, lat, lon, grid_ref: str):
+        """
+            Simple routine to find the nearest J,I coordinates for given lat lon
+            Usage: [J,I] = findJI(49, -12, nav_lat_grid_T, nav_lon_grid_T)
+        """
+
+        interal_lat = f"gphi{grid_ref}"
+        interal_lon = f"glam{grid_ref}"
+        dist2 = xa.ufuncs.square(self.dataset[interal_lat] - lat) + xa.ufuncs.square(self.dataset[interal_lon] - lon)
+        [t, y, x] = np.unravel_index(dist2.argmin(), dist2.shape)
+        return [y, x]
+
+    def transect_indices(self, start: tuple, end: tuple, grid_ref: str = 'T'):
+
+        if len(grid_ref) != 1:
+            raise AssertionError("grid_ref should be either T, V, U, F")
```

Comment:
This should probably be a ValueError. AssertionError is specifically for failed assert statements. Alternatively, use an assert rather than an if statement. https://docs.python.org/3/library/exceptions.html

Reply:
what do you think is the better (more python) approach? if or assert?

Reply:
I'd go with:
assert grid_ref != 1, "grid_ref should be either T, V, U, F"
Simple one line solution.

Reply:
does that assert the len of grid_ref or the value?

Reply:
That's a mistake of mine, it should have been:
```
assert len(grid_ref) != 1, "grid_ref should be either T, V, U, F"
```

Reply:
wonder if this is better, to be honest

 assert type(grid_ref) is str, "grid_ref should be either T, V, U, F"
 assert grid_ref.upper() in ['T', 'V', 'U', 'F'], "grid_ref should be either T, V, U, F"

Reply:
How about:
```
assert isinstance(grid_ref, str) and len(grid_ref) != 1, "grid_ref should be either \"T\", \"V\", \"U\", \"F\""
```

Reply:
was thinking that we could check that the value of grid_ref is within the range we want, rather than just its one character long... 

also wasn't sure what the python best practice is with multiple tests in an assert statement

Reply:
Updated suggestion:
```
assert isinstance(grid_ref, str) and grid_ref.upper() in ("T", "V", "U", "F"), \
    "grid_ref should be either \"T\", \"V\", \"U\", \"F\""
```
Split across two lines because it's now longer.

Reply:
> 
> 
> was thinking that we could check that the value of grid_ref is within the range we want, rather than just its one character long...
> 
> also wasn't sure what the python best practice is with multiple tests in an assert statement

If we were writing unit tests then normally I'd prefer multiple assert statements but in this case  I think the more succinct approach might be better.

But if you still prefer multiple statements then the error message should be a variable rather than two identical literals.

Reply:
Do we need to account for someone passing an empty string?

Reply:
> Do we need to explicitly account for someone passing an empty string?

isn't that already caught by the in the statement?

Reply:
> isn't that already caught by the in the statement?

It should be but an explicit "empty string not allowed"-type message might be better.

Reply:
we already default to T if a user forgets so hopefully, we have the bases covered